src/transformers/models/aria/modular_aria.py

```diff
@@ -0,0 +1,1357 @@
+        self.image_mean = image_mean
+        self.image_std = image_std
+        if split_ratio is None:
+            self.split_ratio = [
+                (1, 2),
+                (1, 3),
+                (1, 4),
+                (1, 5),
+                (1, 6),
+                (1, 7),
+                (1, 8),
+                (2, 4),
+                (2, 3),
+                (2, 2),
+                (2, 1),
+                (3, 1),
+                (3, 2),
+                (4, 1),
+                (4, 2),
+                (5, 1),
+                (6, 1),
+                (7, 1),
+                (8, 1),
+            ]
+        else:
+            self.split_ratio = split_ratio
+
+        self._set_processor_class("AriaProcessor")
+
+    def preprocess(
+        self,
+        images: Union[ImageInput, List[ImageInput]],
+        max_image_size: int = 980,
+        min_image_size: int = 336,
+        return_tensors: Optional[Union[str, TensorType]] = "pt",
+        split_image: Optional[bool] = False,
+        do_convert_rgb: Optional[bool] = True,
+        do_normalize: Optional[bool] = True,
+        resample: PILImageResampling = PILImageResampling.BICUBIC,
+    ):
+        """
+        Process a list of images.
+
+        Args:
+            images (ImageInput or list of ImageInput): The input image or a list of images.
+            max_image_size (int, optional): Maximum image size. Defaults to `self.max_image_size` (980).
+            min_image_size (int, optional): Minimum image size. Defaults to `self.min_image_size` (336).
+            return_tensors (str or TensorType, optional): The type of tensor to return. Defaults to "pt".
+            split_image (bool, optional): Whether to split the image. Defaults to False.
+            do_convert_rgb (bool, optional): Whether to convert the image to RGB. Defaults to True.
+            do_normalize (bool, optional): Whether to normalize the image. Defaults to True.
+            resample (PILImageResampling, optional): The resampling filter to use if resizing the image. Defaults to BICUBIC.
+
+        Returns:
+            BatchFeature: A BatchFeature object containing:
+                - 'pixel_values': Tensor of processed image pixel values.
+                - 'pixel_mask': Boolean pixel mask. This mask is a 2D tensor of shape (max_size, max_size) where:
+                    - True (1) values indicate pixels that belong to the original resized image.
+                    - False (0) values indicate pixels that are part of the padding.
+                  The mask helps distinguish between actual image content and padded areas in subsequent processing steps.
+                - 'num_crops': The maximum number of crops across all images.
+        """
+        max_size = self.max_image_size if max_image_size is None else max_image_size
+        min_size = self.min_image_size if min_image_size is None else min_image_size
+
```

Comment:
can we add validation for images and args here as in 
https://github.com/huggingface/transformers/blob/ae49b218c3d718df90d8e4a109016450fb8f0632/src/transformers/models/clip/image_processing_clip.py#L284-L300 ?

Reply:
Done!